### ISSUE TITLE
Problems with logscale plots

### ISSUE THREAD

**Issue by uleysky** [GenericMappingTools/gmt#4819]
First, not all ticks are drawn inside the frame then --MAP_FRAME_TYPE=inside specified:
```
gmt psbasemap --MAP_FRAME_TYPE=inside -R0.000000000005/1/0.000000000005/1 -JX15cl/15cl -BSWNE -Bya1 -Bxa1 -P -K >test.ps
gmt psbasemap -R0.000000000005/1/0.000000000005/1 -JX15cl/15cl -BSWNE -Bya1 -Bxa1 -P -O >>test.ps
```
![test](https://user-images.githubusercontent.com/5275343/108145502-41611500-7117-11eb-8656-d1397be34649.png)
Custom annotations are cropped too.

Second, the documentation for the -B option contains the following lines: 
```
2. An l is appended to the tickinfo string. Then, log10 of the tick value is plotted at every integer log10 value.
3. A p is appended to the tickinfo string. Then, annotations appear as 10 raised to log10 of the tick value.
```
but tickinfo is not defined anywhere. Same for the paragraph on power axes.

**Comment by PaulWessel** [comment]
The missing annotations are tied to this master-stroke that I implemented some years ago:

`if (GMT->current.setting.map_frame_type & GMT_IS_INSIDE && (fabs (knots[i] - val0) < GMT_CONV8_LIMIT || fabs (knots[i] - val1) < GMT_CONV8_LIMIT)) continue;	/* Skip annotation on edges when MAP_FRAME_TYPE = inside */`

which clearly did not consider log-axes.  These values need to be projected before doing this sort of check.

**Comment by PaulWessel** [comment]
Fixed via #4825 and #4826.
